src/init.ts

```diff
@@ -36,18 +52,29 @@ export async function init() {
         token = process.env.GITHUB_API_TOKEN ?? process.env.FOCUS_DT_GITHUB_API_TOKEN ?? process.env.AUTH_TOKEN;
     }
 
-    let credential: GitCredential | GitUrlCredential | undefined;
+    const userSuppliedToken = !!token;
+
+    token ??= ghAuthToken();
+    let status = await ghAuthStatus(token);
+    if (!status.authenticated) {
+        checkGhInstalled();
+        token = ghAuthLogin();
+        status = await ghAuthStatus(token);
+        checkAuthenticated(status);
+    }
+    if (userSuppliedToken) {
+        checkRequiredScopes(status);
+    }
+    else if (!status.scopes?.includes("read:project")) {

```

Comment:
I think this is the same as `checkRequiredScopes` below.

Reply:
Ah, same check, but that one throws.

Reply:
I guess it can't be as easy the header returning the full set of scopes.